src/html-purify.js

```diff
@@ -53,14 +53,56 @@ See the accompanying LICENSE file for terms.
         return false;
     }
 
+    function addQuoteToValue(action, value) {
+        if (action & derivedState.TransitionName.DQ_ATTR) {         // collect double quoted value
+            return '"' + value + '"';
+        } 
+        else if (action & derivedState.TransitionName.SQ_ATTR) {    // collect single quoted value
+            return "'" + value + "'";
+        } 
+        else /*if (action & derivedState.TransitionName.UQ_ATTR) {  // collect unquoted value */{
+            return value;
+        }
+    }
+
     function processTransition(prevState, nextState, i) {
         /* jshint validthis: true */
         /* jshint expr: true */
         var parser = this.parser,
+            action = derivedState.Transitions[prevState][nextState],
             idx, tagName, attrValString, openedTag, key, value;
 
-        
-        switch (derivedState.Transitions[prevState][nextState]) {
+        // check if tag and/or attr is available for collection
+        // collect the values (if any) only when the attr name is allowed
+        if (action & 0xF && contains(this.attributesWhitelist, (key = parser.getAttributeName()))) {
```

Comment:
:+1: gd pt. it's really open for discussion.

first of all, a disallowed tag won't get emitted to the output. so, this is not a security concern.

it's then a decision based on the performance. 
i actually tried to further constraint this by `contains(this.tagsWhitelist, tagName)` and benchmarked that. the performance is really subject to how many tags are disallowed. 
given the sample `1m.html`, it has gd tags more than bad tags (and i assume most emails are like that as in the Mail example, and only those using bad tags will be perf. penalized), i therefore dropped it.

in the future, i anticipate tag-specific attribute whitelisting (e.g., `src` is allowed only for some specific tags). this is where a hash map or function will take care of the tag+attr whitelisting.

so still want to constraint this?